web-server/src/utils/logFormatter.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  ParsedLog,
+  generalLogRegex,
+  httpLogRegex,
+  redisLogRegex,
+  postgresLogRegex,
+  postgresMultiLineLogRegex,
+  dataSyncLogRegex,
+  validLogLevels
+} from '../constants/log-formatter';
+
+export const parseLogLine = (rawLogLine: string): ParsedLog | null => {
+  const generalLogMatch = rawLogLine.match(generalLogRegex);
+  if (generalLogMatch) {
+    const [, timestamp, , logLevel, message] = generalLogMatch;
+    return {
+      timestamp,
+      logLevel,
+      message
+    };
+  }
+
+  const httpLogMatch = rawLogLine.match(httpLogRegex);
+  if (httpLogMatch) {
+    const [, ip, , , timestamp, request, status, bytes, referer, userAgent] =
+      httpLogMatch;
+    const [method, path] = request.split(' ');
+    return {
+      timestamp,
+      logLevel: 'INFO', // Assuming all HTTP logs are INFO level
```

Comment:
all these string constants should be wrapped in an enum in our `src/types/resources.ts`
```ts
enum YourEnumName 
{ 
  INFO = 'INFO', 
  DEBUG = 'DEBUG'
  ...other_possible_values... 
}
```

Reply:
done , created a enum LogLevel.